packages/@uppy/xhr-upload/src/index.ts

```diff
@@ -117,7 +114,16 @@ const defaultOptions = {
   withCredentials: false,
   responseType: '',
   getResponseData(responseText) {
-    return JSON.parse(responseText)
+    let parsedResponse = {}
+    try {
+      parsedResponse = JSON.parse(responseText)
+    } catch {
+      // ignore
+    }
+    // We don't have access to the B (Body) generic here
+    // so we have to cast it to any. The user facing types
+    // remain correct, this is only to please the merging of default options.
+    return parsedResponse as any
```

Comment:
I don't think an empty object is a correct response here, `undefined` would be more appropriate, no?
```suggestion
    try {
      return JSON.parse(responseText)
    } catch {
      // ignore
    }
```
or 
```suggestion
    if (responseText) JSON.parse(responseText)
```

Reply:
It's a regression because it was used return `{}`, until I removed it in https://github.com/transloadit/uppy/pull/4892. So `{}` is what is has to be for now.